reddit_service_websockets/socketserver.py

```diff
@@ -61,8 +81,47 @@ def upgrade_connection(self):
             return ["Forbidden"]
 
         self.environ["signature_validated"] = True
+
+        # Check if compression is supported.  The RFC explanation for the
+        # variations on what is accepted here is convoluted, so we'll just
+        # stick with the happy case:
+        #
+        #    https://tools.ietf.org/html/rfc6455#page-48
+        #
+        # Worst case scenario, we fall back to not compressing.
+        extensions = self.environ.get('HTTP_SEC_WEBSOCKET_EXTENSIONS')
+        extensions = {extension.split(";")[0].strip()
+                      for extension in extensions.split(",")}
+        self.environ["supports_compression"] = \
+            "permessage-deflate" in extensions
+
         return super(WebSocketHandler, self).upgrade_connection()
 
+    def start_response(self, status, headers, exc_info=None):
+        # Ideally, this would probably go in `upgrade_connection`, but we have
+        # no way to modify the headers there without duplicating the logic of
+        # the entire method.  This is a much smaller entry point.
+        # Ideally, geventwebsocket would just support this stuff.
```

Comment:
since we've got monkeypatches already, would this be something we could hack into it?